docs/usage.md

```diff
@@ -180,30 +303,33 @@ cachi2 fetch-deps --source ./atomic-reactor '{
 }'
 ```
 
-Details: [pre-fetch dependencies](#pre-fetch-dependencies)
+#### Generate environment variables (pip)
 
-### Generate environment variables (pip)
+Next, we need to generate the environment file so that the `pip install` command can find the cached dependencies
 
 ```shell
 cachi2 generate-env ./cachi2-output -o ./cachi2.env --for-output-dir /tmp/cachi2-output
 ```
 
+We can see the variables needed by the package manager:
+
 ```shell
 $ cat cachi2.env
 export PIP_FIND_LINKS=/tmp/cachi2-output/deps/pip
 export PIP_NO_INDEX=true
 ```
 
-Details: [generate environment variables](#generate-environment-variables)
+#### Inject project files (pip)
 
-### Inject project files (pip)
+In order to be able to install pip dependencies in a hermetic environment, we need to perform the injection to
+change the remote dependencies to instead point to the local file system.
```

Comment:
Might be worth calling out that this is only needed for [external dependencies](https://github.com/containerbuildsystem/cachi2/blob/main/docs/pip.md#external-dependencies). If all your deps come from PyPI, cachi2 won't replace anything